### PR TITLE
issue/1804: migrate some theme less into core

### PR DESCRIPTION
**Pull request by oliverfoster**

### CODE REVIEW

File: grunt/config/less.js [hunk 1]
```diff
@@ -4,12 +4,17 @@ module.exports = function (grunt, options) {
             options:{
                 baseUrl: '<%= sourcedir %>',
                 mandatory: [
-                    '<%= sourcedir %>core/less/**/*.less'
+                    '<%= sourcedir %>core/less/*.less',
```
Comment (oliverfoster) [inline grunt/config/less.js]: is there any need for these to be listed individually? they produce the same thing, only the original is shorter.

File: src/core/less/browser/legacy.less [hunk 2]
```diff
@@ -1,3 +1,19 @@
+// Legacy
```
Comment (oliverfoster) [inline src/core/less/browser/legacy.less]: this whole file can go, we no longer support ie8 in upcoming v3

File: src/core/less/common/accessibility.less [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+.aria-label {
+    position:absolute !important;
+    &.relative {
+        position:relative !important;
+    }
+    left:0px !important;
+    width:auto !important;
+    height:auto !important;
+    overflow:auto !important;
+    filter: progid:DXImageTransform.Microsoft.Alpha(Opacity=0); /*IE8*/
```
Comment (oliverfoster) [inline src/core/less/common/accessibility.less]: this line can go, it's ie8 only

File: src/core/less/common/accessibility.less [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+    }
+
+/*HIDDEN FOCUS GUARD*/
+#a11y-focusguard {
+    left:0px !important;
+    bottom:0px !important;
+    width:auto !important;
+    height:auto !important;
+    overflow:auto !important;
+    filter: progid:DXImageTransform.Microsoft.Alpha(Opacity=0); /*IE8*/
```
Comment (oliverfoster) [inline src/core/less/common/accessibility.less]: this line can go, ie8 only

File: src/core/less/common/accessibility.less [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+
+/*HIDDEN FOCUS/ALERT ELEMENT */
+#a11y-selected, #a11y-selected * {
+    position:fixed !important;
+    left:0px !important;
+    bottom:0px !important;
+    width:auto !important;
+    height:auto !important;
+    overflow:auto !important;
+    filter: progid:DXImageTransform.Microsoft.Alpha(Opacity=0); /*IE8*/
```
Comment (oliverfoster) [inline src/core/less/common/accessibility.less]: this line can go, ie8 only

File: src/core/less/common/accessibility.less [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+
+/*FOCUS DISTRACTION*/
+#a11y-focuser {
+    position:fixed !important;
+    left:0px !important;
+    top:0px !important;
+    width:auto !important;
+    height:auto !important;
+    overflow:auto !important;
+    filter: progid:DXImageTransform.Microsoft.Alpha(Opacity=0); /*IE8*/
```
Comment (oliverfoster) [inline src/core/less/common/accessibility.less]: this line can go, ie8 only

File: src/core/less/common/accessibility.less [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+    background: rgba(0,0,0,0) !important;
+    font-size: 1px !important;
+    padding:0 !important;
+    margin:0 !important;
+    line-height: normal !important;
+    z-index:1;
+}
+
+.no-select {
+    -webkit-touch-callout: none;
```
Comment (oliverfoster) [inline src/core/less/common/accessibility.less]: .no-select, .visibility-hidden and .display-none shouldn't be in here, they're not for accessibility

File: src/core/less/common/buttons.less [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+            border-color: @button-border-color-disabled;
+        }
+    }
+
+    &.disabled, &:disabled { .disabled-style; }
+    // IE8 fix needed to be in its own line
+    &[disabled="disabled"] { .disabled-style; }
+}
+
+.buttons {
```
Comment (oliverfoster) [inline src/core/less/common/buttons.less]: this class down is actually for the question component buttons and should be moved to the modules folder under "buttons.less"

File: src/core/less/structure/article.less [hunk 9]
```diff
@@ -0,0 +1,53 @@
+// Article styling
+// ===============
+
+.article {
+    background-color: @background-color;
+
+    .article-inner {
+        padding: @article-padding-top 0 @article-padding-bottom;
+        @media all and (max-width: (@device-width-large + 20px)) {
```
Comment (oliverfoster) [inline src/core/less/structure/article.less]: all of the media queries like the ones in this file are additional styling for a specific theme specific look, and should be removed

File: src/core/less/structure/block.less [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+.block {
+    background-color: @background-color;
+    position: relative;
+
+    .block-inner {
+        max-width: @device-width-large;
+        margin: auto;
+        padding: @block-padding-top 0 @block-padding-bottom;
+        // reduce padding as screen width decreases
+        @media all and (max-width: (@device-width-large + 20px)) {
```
Comment (oliverfoster) [inline src/core/less/structure/block.less]: all of the media queries like the ones in this file are additional styling for a specific theme specific look, and should be removed

File: src/core/less/structure/component.less [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+.component-container { zoom: 1; }
+
+.component-left {
+    width:100%;
+    float:none;
+    @media all and (min-width:(@device-width-medium + 1px)) {
+        width:50%;
+        float:left;
+        .component-inner {
+            margin-right:5%;
```
Comment (oliverfoster) [inline src/core/less/structure/component.less]: remove these theme specific margins

File: src/core/less/structure/page.less [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+    background-color: @background-color;
+
+    .page-header {
+        background-color: @background-color-inverted;
+    }
+
+    .page-header-inner {
+        padding: @page-padding-top 0 @page-padding-bottom;
+
+        @media all and (max-width: (@device-width-large + 20px)) {
```
Comment (oliverfoster) [inline src/core/less/structure/page.less]: all of the media queries like the ones in this file are additional styling for a specific theme specific look, and should be removed

File: src/core/less/structure/wrapper.less [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+}
+
+
+// @TODO: Theme migration
+#wrapper {
+    width: 100%;
+    margin: auto;
+    overflow-x: hidden;
+    padding-top: (@navigation-padding * 2) + @icon-size;
+    .ie8-max-width;
```
Comment (oliverfoster) [inline src/core/less/structure/wrapper.less]: remove ie8 only bits

File: src/core/less/variables/animations.less [hunk 14]
```diff
@@ -0,0 +1,32 @@
+// Animations
```
Comment (oliverfoster) [inline src/core/less/variables/animations.less]: do we need a vote on whether this is core?

File: src/core/less/variables/borders.less [hunk 15]
```diff
@@ -0,0 +1,29 @@
+// Borders and Outlines
+// ====================
+
+@focus-outline: 3px solid orange; //focus outline is displayed when accessibility is enabled
```
Comment (oliverfoster) [inline src/core/less/variables/borders.less]: accessibility or borders?

File: src/core/less/variables/colors.less [hunk 16]
```diff
@@ -0,0 +1,220 @@
...
+@validation-error-border-color: @validation-error;
+
+// ======================
+
+// Disabled States
+@disabled-color: #727272;
+@disabled-color-inverted: #ccc;
+
+@transparency: rgba(255,255,255,0.8);
+@transparency-fallback-png: url('assets/transparency-light.png');
```
Comment (oliverfoster) [inline src/core/less/variables/colors.less]: ie8 only

File: src/core/less/variables/colors.less [hunk 17]
```diff
@@ -0,0 +1,220 @@
+// Main Colors
```
Comment (oliverfoster) [inline src/core/less/variables/colors.less]: i don't know whether the "module" specific stuff from files like this should be broken up into "modules/notify/colors.less" etc

File: src/core/less/variables/devices.less [hunk 18]
```diff
@@ -0,0 +1,13 @@
...
+@device-width-small: unit(@adapt-device-small, px);
+@device-width-medium: unit(@adapt-device-medium, px);
+@device-width-large: unit(@adapt-device-large, px);
+
+
+// Max-widths
+// ==========
+
+@max-width: 1400px;
+@ie8-max-width: 1024px;
```
Comment (oliverfoster) [inline src/core/less/variables/devices.less]: ie8 only

File: src/core/less/variables/responsive.less [hunk 19]
```diff
@@ -0,0 +1,30 @@
+// =================
```
Comment (oliverfoster) [inline src/core/less/variables/responsive.less]: is this file still needed?

File: src/core/less/variables/spacings.less [hunk 20]
```diff
@@ -0,0 +1,259 @@
+// Spacings setup
```
Comment (oliverfoster) [inline src/core/less/variables/spacings.less]: i think all of these variables should be set to 0 (except where necessary for bare structure)

File: src/core/less/variables/spacings.less [hunk 21]
```diff
@@ -0,0 +1,259 @@
...
+@body-text-margin: 30px;
+@instruction-text-margin: 14px;
+@button-margin: 30px;
+
+
+// Paddings - Detailed Spacings setup
+// ==================================
+
+// Item padding
+@item-padding-top: @item-padding;
```
Comment (oliverfoster) [inline src/core/less/variables/spacings.less]: these type of secondary relationships should be broken. a theme will want to/should do these differently. set all to 0.

there should be no or few hierarchical abstractions in the core theme. such that x variable has y impact. sweet and simple. easy to create relationships later.

File: grunt/config/less.js (hunk 1 above)
Comment (nilslp) [reply grunt/config/less.js]: Nope, updating.

File: src/core/less/common/accessibility.less (hunk 3 above)
Comment (nilslp) [reply src/core/less/common/accessibility.less]: Done.

File: src/core/less/common/accessibility.less (hunk 4 above)
Comment (nilslp) [reply src/core/less/common/accessibility.less]: Done.

File: src/core/less/common/accessibility.less (hunk 5 above)
Comment (nilslp) [reply src/core/less/common/accessibility.less]: Done.

File: src/core/less/common/accessibility.less (hunk 6 above)
Comment (nilslp) [reply src/core/less/common/accessibility.less]: Done.

File: src/core/less/common/accessibility.less (hunk 7 above)
Comment (nilslp) [reply src/core/less/common/accessibility.less]: Done.

File: src/core/less/structure/article.less (hunk 9 above)
Comment (nilslp) [reply src/core/less/structure/article.less]: Done.

File: src/core/less/structure/block.less (hunk 10 above)
Comment (nilslp) [reply src/core/less/structure/block.less]: Done.

File: src/core/less/structure/component.less (hunk 11 above)
Comment (nilslp) [reply src/core/less/structure/component.less]: Done.

File: src/core/less/structure/page.less (hunk 12 above)
Comment (nilslp) [reply src/core/less/structure/page.less]: Done.

File: src/core/less/structure/wrapper.less (hunk 13 above)
Comment (nilslp) [reply src/core/less/structure/wrapper.less]: Done.

File: src/core/less/variables/animations.less (hunk 14 above)
Comment (nilslp) [reply src/core/less/variables/animations.less]: If your wondering about it, its probably good to go.
Animation is a nicety that developers can do on their own theme I guess.
Thanks, removed.

File: src/core/less/variables/borders.less (hunk 15 above)
Comment (nilslp) [reply src/core/less/variables/borders.less]: Accessibility. Done.

File: src/core/less/variables/colors.less (hunk 16 above)
Comment (nilslp) [reply src/core/less/variables/colors.less]: Done.

File: src/core/less/variables/devices.less (hunk 18 above)
Comment (nilslp) [reply src/core/less/variables/devices.less]: Done.

File: src/core/less/variables/spacings.less (hunk 20 above)
Comment (nilslp) [reply src/core/less/variables/spacings.less]: Done.

File: src/core/less/variables/spacings.less (hunk 21 above)
Comment (oliverfoster) [reply src/core/less/variables/spacings.less]: obviously, this might not be possible.

File: src/core/less/common/buttons.less (hunk 8 above)
Comment (nilslp) [reply src/core/less/common/buttons.less]: Done.

File: src/core/less/variables/responsive.less (hunk 19 above)
Comment (nilslp) [reply src/core/less/variables/responsive.less]: Removed. Done.

Comment (nilslp) [reply src/core/less/variables/responsive.less]: This one I'm not sure about?

Based on the rest of the changes I would say no, but based on the conversation of should the framework build a course that looks horrible and in this comments case unresponsive.

File: src/core/less/variables/colors.less (hunk 17 above)
Comment (nilslp) [reply src/core/less/variables/colors.less]: Done.

**Comment by nilslp** [comment]
Hey @oliverfoster how is this looking now?

**Comment by oliverfoster** [comment]
closed as work stream has been superseded
sorry @nilslp
